src/markup.js

```diff
@@ -0,0 +1,269 @@
+      </tr>
+      <tr>
+         <th>Aborted:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._aborted}</td>
+      </tr>
+      <tr>
+         <th>Inconclusive:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._inconclusive}</td>
+      </tr>
+      <tr>
+         <th>PassedButRunAborted:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._passedButRunAborted}</td>
+      </tr>
+      <tr>
+         <th>NotRunnable:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._notRunnable}</td>
+      </tr>
+      <tr>
+         <th>NotExecuted:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._notExecuted}</td>
+      </tr>
+      <tr>
+         <th>Disconnected:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._disconnected}</td>
+      </tr>
+      <tr>
+         <th>Warning:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._warning}</td>
+      </tr>
+      <tr>
+         <th>Completed:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._completed}</td>
+      </tr>
+      <tr>
+         <th>InProgress:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._inProgress}</td>
+      </tr>
+      <tr>
+         <th>Pending:</th>
+         <td>${testData.TrxData.TestRun.ResultSummary.Counters._pending}</td>
+      </tr>
+    </table>
+  </details>
+  `;
+}
+
+function getTestResultsMarkup(testData) {
+  let resultsMarkup = '';
+  if (testData.IsEmpty) {
+    return getNoResultsMarkup(testData);
+  } else {
+    const unittests = testData.TrxData.TestRun.TestDefinitions.UnitTest;
+    if (Array.isArray(unittests)) {
+      for (const data of unittests) {
+        resultsMarkup += getSingleTestMarkup(data, testData);
+      }
+      return resultsMarkup.trim();
+    } else {
+      return getSingleTestMarkup(unittests, testData);
+    }
+  }
+}
+
+function getNoResultsMarkup(testData) {
+  const runInfo = testData.TrxData.TestRun.ResultSummary.RunInfos.RunInfo;
+  const testResultIcon = getTestOutcomeIcon(runInfo._outcome);
+  const resultsMarkup = `
+  <details>
+    <summary>${testResultIcon} ${runInfo._computerName}</summary> 
+    <table>
+      <tr>
+        <th>Run Info</th>
+        <td><code>${runInfo.Text}</code></td>
+      </tr>
+    </table>      
+    </details>
+  `;
+  return resultsMarkup;
+}
+
+function getTestOutcomeIcon(testOutcome) {
+  if (testOutcome === 'Passed') return ':heavy_check_mark:';
+  if (testOutcome === 'Failed' || testOutcome === 'Error') return ':x:';
+  if (testOutcome === 'NotExecuted') return ':radio_button:';
+
+  return ':grey_question:';
+}
+
+function getSingleTestMarkup(data, testData) {
+  core.debug(`Processing ${data._name}`);
+
+  let resultsMarkup = '';
+  const testResult = getUnitTestResult(data._id, testData.TrxData.TestRun.Results);
+  if (testResult && testResult._outcome === 'Failed') {
```

Comment:
I'm going to actually change this up a bit and pull the failing test logic into the method that calls this. 

I'm pretty sure it is this way because of how it used to work when they reported all test results.  Because of some size limitations in the status check they had to cut it down to just the failing tests.

Reply:
Oh yeah, I could definitely see that.